src/VirtualClient/VirtualClient.Actions/ScriptExecutor/PowershellExecutor.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Actions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Logging;
+    using VirtualClient.Common;
+    using VirtualClient.Common.Telemetry;
+    using VirtualClient.Contracts;
+
+    /// <summary>
+    /// The Generic Script executor for Python
+    /// </summary>
+    public class PowershellExecutor : ScriptExecutor
+    {
+        /// <summary>
+        /// Constructor for <see cref="PowershellExecutor"/>
+        /// </summary>
+        /// <param name="dependencies">Provides required dependencies to the component.</param>
+        /// <param name="parameters">Parameters defined in the profile or supplied on the command line.</param>
+        public PowershellExecutor(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters)
+             : base(dependencies, parameters)
+        {
+        }
+
+        /// <summary>
+        /// Executes the workload.
+        /// </summary>
+        protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            using (BackgroundOperations profiling = BackgroundOperations.BeginProfiling(this, cancellationToken))
+            {
+                using (IProcessProxy process = await this.ExecuteCommandAsync(
+                    "powershell",
+                    $"-NoProfile -Command \"cd '{this.WorkloadPackage.Path}';{this.ExecutablePath} {this.CommandLine}\"",
```

Comment:
You might already did, just checking if we need any of these flags if we are logging in as session 0 in Windows 
-WindowStyle Hidden -NonInteractive -ExecutionPolicy

Reply:
Added these flags. Thanks for pointing out.

Reply:
You don't have to add these but these are just for investigation, add only if we need them

Reply:
I had checked out these arguments individually, and all seemed fit to be added. Hence, added them all. Also, added a line in documentation to use this in Elevated mode.